ccb.go

```diff
@@ -0,0 +1,299 @@
+// depending on the response from CCB, we may want to reformat the structs before passing them back to the user.
+type responseHandler func(ctx iris.Context, response CCBResponse)
+
+// makeCCBRequest can currently handle
+func makeCCBRequest(ctx iris.Context, url string, method string, handler responseHandler) {
+	// build http request
+	client := &http.Client{}
+	req, err := http.NewRequest(method, url, nil)
+	if err != nil {
+		fmt.Println(err) // TODO: this should be logged
+		ctx.StatusCode(http.StatusInternalServerError)
+		ctx.WriteString("Error: Unable to build request to database. " + err.Error())
+		return
+	}
+	req.SetBasicAuth(os.Getenv("CCB_USERNAME"), os.Getenv("CCB_PASSWORD"))
+
+	// send request
+	resp, err := client.Do(req)
+	if err != nil {
+		fmt.Println(err) // TODO: this should be logged
+		ctx.StatusCode(http.StatusInternalServerError)
+		ctx.WriteString("Error: Unable to send request to database. " + err.Error())
+		return
+	}
+
+	// TODO: Error handling for error responses from CCB
+	defer resp.Body.Close()
+
+	var data CCBResponse
+
+	respBody, _ := ioutil.ReadAll(resp.Body)
+	err = xml.Unmarshal(respBody, &data)
+	if err != nil {
+		fmt.Println(err) // TODO: this should be logged
+		ctx.StatusCode(http.StatusInternalServerError)
+		ctx.WriteString("Error unmarshalling response from database")
+		return
+	}
+
+	handler(ctx, data)
+}
+
+func whoIsResponseHandler(ctx iris.Context, resp CCBResponse) {
+	jsonResponse, err := json.Marshal(resp)
+	if nil != err {
+		fmt.Println(err) // TODO: this should be logged
+		ctx.StatusCode(http.StatusInternalServerError)
+		ctx.WriteString("Error marshalling to JSON")
+		return
+	}
+
+	// write back the body as JSON
+	ctx.Write([]byte(jsonResponse))
+}
+
+// formResponseHandler changes the structure of the CCB response data to a more readable structure
+// before passing the JSON back to the user.
+func formResponseHandler(ctx iris.Context, resp CCBResponse) {
+	// fill in count field from CCBResponse to a FormResponses struct
+	// create variables for iterator to fill in.
+	var formResponses FormResponses
+	formResponses.Count = resp.Response.FormResponses.Count
+	var jsonResponse []byte
+	var err error
+
+	// if there are more than 0 form responses returned, fill in the FormResponses struct
+	if formResponses.Count != "0" {
+		// range over the form responses from CCB
+		for _, v := range resp.Response.FormResponses.FormResponse {
+			profInfo := map[string]string{} // this will contain profile information
+			answers := map[string]string{}  // this will contain the form questions and answers
+
+			// range over profile information and move to a map with info.Name as the key and info.Text as the value
+			for _, info := range v.ProfileFields.ProfileInfo {
+				profInfo[info.Name] = info.Text
+			}
+
+			// range over XML unmarshalled "Answers" and move form questions and answers to a map
+			for i, t := range v.Answers.Title {
```

Comment:
can we name t with what the value represents? unsure of what's actually in that value

Reply:
haha! yeah that was definitely a shortcut. Will change to questionTitle or something.